pkg/generators/markers.go

```diff
@@ -0,0 +1,242 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generators
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"k8s.io/gengo/types"
+	"k8s.io/kube-openapi/pkg/validation/spec"
+)
+
+// Subset of spec.SchemaProps that may be placed into a marker comment
+type ValueValidations struct {
+	Maximum          *float64 `json:"maximum,omitempty"`
+	ExclusiveMaximum *bool    `json:"exclusiveMaximum,omitempty"`
+	Minimum          *float64 `json:"minimum,omitempty"`
+	ExclusiveMinimum *bool    `json:"exclusiveMinimum,omitempty"`
+	MaxLength        *int64   `json:"maxLength,omitempty"`
+	MinLength        *int64   `json:"minLength,omitempty"`
+	Pattern          *string  `json:"pattern,omitempty"`
+	MaxItems         *int64   `json:"maxItems,omitempty"`
+	MinItems         *int64   `json:"minItems,omitempty"`
+	UniqueItems      *bool    `json:"uniqueItems,omitempty"`
+	MultipleOf       *float64 `json:"multipleOf,omitempty"`
+	MaxProperties    *int64   `json:"maxProperties,omitempty"`
+	MinProperties    *int64   `json:"minProperties,omitempty"`
+}
+
+type CommentTags struct {
```

Comment:
What's our plan with existing supported comments like +default and +nullable? Is the plan for all tags to adhere to the +k8s:validation:<field> convention?

Reply:
Thanks for raising that point, I don't think we want it AT ALL.

Reply:
+default and +nullable won't be modified in any way. The prefixes are only being applied to the comment types added in the current PR

Reply:
prefixes have been removed